modules/lib/file-overlays.nix

```diff
@@ -0,0 +1,22 @@
+{ pkgs, config }:
+
+# Overlay functions are applied in the order of the list.
+# Latter function will be applied latter and get more precedance.
+[
+  (self: super:
+    pkgs.lib.attrsets.optionalAttrs
+    (with config.programs.blesh; enable && manageBashrc && super ? ".bashrc") {
+      ".bashrc" = super.".bashrc" // {
+        source = pkgs.writeTextFile {
+          inherit (super.".bashrc".source) name;
+          text = ''
+            [[ $- == *i* ]] && source '${config.programs.blesh.package}/share/ble.sh' --noattach
```

Comment:
I was waiting for https://github.com/NixOS/nixpkgs/pull/185866 made by @tars0x9752 to be processed first, but it seems to take time to finally merge it, so let me write this now. As I have commented in the review comments below https://github.com/NixOS/nixpkgs/pull/181963#pullrequestreview-1056052774, the current install location is wrong. Currently, all the related files are put directly under `$prefix/share` (such as `$prefix/share/ble.sh`, `$prefix/share/keymap`, and `$prefix/share/lib`), but all these files should be placed under the directory `$prefix/share/blesh` such as `$prefix/share/blesh/ble.sh`).

Could you first process them including other comments below https://github.com/NixOS/nixpkgs/pull/181963#pullrequestreview-1056052774 (probably after concluding https://github.com/NixOS/nixpkgs/pull/185866) before proceeding with this PR? Also, it would be great if you could edit [the Wiki page](https://github.com/akinomyoga/ble.sh/wiki/Manual-A1-Installation#user-content-nixpkgs) after that.

Reply:
Whoops, I thought the wrong installation path was already fixed on https://github.com/NixOS/nixpkgs/pull/185866 because `installFlags` attribute was removed there, and I completely forgot to fix them in this PR. But as this PR passes the tests without fixing the path, the installation path was actually not fixed yet.

I'll see why the current path is still being wrong without `installFlags` attribute, and also fix the path in this PR. Before these investigation this PR shouldn't be merged, so I'm converting it to draft.

Reply:
Thank you for your reply!

> Whoops, I thought the wrong installation path was already fixed on [NixOS/nixpkgs#185866](https://github.com/NixOS/nixpkgs/pull/185866) because `installFlags` attribute was removed there,

I think the installation path isn't fixed by NixOS/nixpkgs#185866 as far as I see lines [[1]](https://github.com/NixOS/nixpkgs/pull/185866/files#diff-a321819ac826b9eccaf69268d655b5826335e81f73da8d822bcd33efaf801ab9R38) and [[2]](https://github.com/NixOS/nixpkgs/pull/185866/files#diff-a321819ac826b9eccaf69268d655b5826335e81f73da8d822bcd33efaf801ab9L41). I have been thinking that we could process this in a separate PR than NixOS/nixpkgs#185866, but now the merging process turned out to be very slow. If @tars0x9752 could agree, now I'm becoming to think we can bundle the fix of the installation path with NixOS/nixpkgs#185866 although it might not be the original purpose of that PR.

<!-- I tried to understand what `fetchzip` does but not sure what is done in this line: https://github.com/NixOS/nixpkgs/blob/348518c7ef7a07ca17cdf2e4c090573e24f7d858/pkgs/build-support/fetchzip/default.nix#L64. Maybe `out` has a different value for fetch implementations. -->

Reply:
Ah, I remembered that the PR removes build process and installs ble.sh by just copying the built scripts. So the [line [1]](https://github.com/NixOS/nixpkgs/pull/185866/files#diff-a321819ac826b9eccaf69268d655b5826335e81f73da8d822bcd33efaf801ab9R38) should be moved to `$out/share/blesh` instead of `$out/share`.

I agree that the path is better to be fixed on the PR.